workers/unity/Packages/com.improbable.gdk.tools/DownloadCoreSdk.cs

```diff
@@ -136,6 +150,81 @@ private static DownloadResult Download()
+                foreach (var pluginPath in pluginPaths)
+                {
+                    var plugin = AssetImporter.GetAtPath(AssetDatabase.GUIDToAssetPath(pluginPath)) as PluginImporter;
+                    if (!plugin)
+                    {
+                        continue;
+                    }
+
+                    if (pluginDirectoryCompatibility.AnyPlatformCompatible)
+                    {
+                        plugin.SetCompatibleWithAnyPlatform(true);
+                        if (pluginDirectoryCompatibility.IncompatiblePlatforms != null)
+                        {
+                            foreach (var buildTarget in pluginDirectoryCompatibility.IncompatiblePlatforms)
+                            {
+                                plugin.SetExcludeFromAnyPlatform(buildTarget, true);
+                            }
+                        }
+
+                        if (pluginDirectoryCompatibility.CompatiblePlatforms != null)
+                        {
+                            Debug.LogError($"Bad plugin compatibility configuration for  {pluginDirectoryCompatibility.Path}. Compatible with any shouldn't be used with a list of compatible platforms");
+                        }
+                    }
+                    else
+                    {
+                        plugin.SetCompatibleWithAnyPlatform(false);
+                        if (pluginDirectoryCompatibility.CompatiblePlatforms != null)
+                        {
+                            foreach (var buildTarget in pluginDirectoryCompatibility.CompatiblePlatforms)
+                            {
+                                plugin.SetCompatibleWithPlatform(buildTarget, true);
+                            }
+                        }
+
+                        if (pluginDirectoryCompatibility.IncompatiblePlatforms != null)
+                        {
+                            Debug.LogError($"Bad plugin compatibility configuration for  {pluginDirectoryCompatibility.Path}. Not compatible with any shouldn't be used with a list of incompatible platforms");
+                        }
+                    }
+
+                    plugin.SetCompatibleWithEditor(pluginDirectoryCompatibility.EditorCompatible);
+
+                    plugin.SaveAndReimport();
+                }
+            }
+        }
+
+        private class PluginDirectoryCompatibility
+        {
+            public PluginDirectoryCompatibility(string path,
+                bool anyPlatformCompatible,
+                List<BuildTarget> compatiblePlatforms,
+                List<BuildTarget> incompatiblePlatforms,
+                bool editorCompatible)
+            {
+                Path = path;
+                AnyPlatformCompatible = anyPlatformCompatible;
+                CompatiblePlatforms = compatiblePlatforms;
+                IncompatiblePlatforms = incompatiblePlatforms;
+                EditorCompatible = editorCompatible;
+            }
```

Comment:
If you do validation of the arguments here - you can pull a lot of the nested logic out from above into here. Logically - the validation should happen here as well imo

Reply:
Good point, let me rework that